src/redux/rockets/Rockets.js

```diff
@@ -0,0 +1,54 @@
+import getRockets from './RocketAPI';
+
+const GET_ROCKET = 'rocketStore / rockets / GET_ROCKET';
+const ADD_RESERVARION = 'rocketStore / rockets / ADD_RESERVARION';
+
+const initialState = [];
+
+export const loadRocket = (payload) => ({
+  type: GET_ROCKET,
+  payload,
+});
+
+export const addReservation = (payload) => ({
+  type: ADD_RESERVARION,
+  payload,
+});
+
+export const loadRockets = () => (dispatch) => {
+  getRockets().then((rockets) => {
+    const rocketData = [];
+    rockets.forEach((rocket) => {
+      const rocketId = {};
+      rocketId.id = rocket.rocket_id;
+      rocketId.name = rocket.rocket_name;
+      rocketId.description = rocket.description;
+      rocketId.image = rocket.flickr_images;
+      rocketId.reserved = false;
+      rocketData.push(rocketId);
+    });
+    dispatch(loadRocket(rocketData));
+  });
+};
+
+const reducer = (state = initialState, action) => {
+  switch (action.type) {
+    case GET_ROCKET:
+      return action.payload;
+
+    case ADD_RESERVARION: {
+      const newState = state.map((rocket) => {
+        if (rocket.id !== action.payload) {
+          return rocket;
+        }
+        rocket.reserved = !rocket.reserved;
+        return rocket;
+      });
+      return newState;
+    }
+    default:
+      return state;
+  }
+};
+
+export default reducer;
```

Comment:
- [ ] Even though the implementation is working perfectly, a bunch of code is still in one file. It's not helpful at all. Let us split the code into separate files so that we can make the component smaller.

      -  Let us create three folders under the redux folder
      
- Actions folder will keep the `action creators and action types`   

Ex of action creators: `export const allowLeaveMission = (payload) => ({
  type: LEAVE_MISSION,
  payload,
});   `

Example of action type: `const JOIN_MISSION = 'space/missions/JOIN_MISSION';`

- Reducers folder will keep the cases 

Example:

`const missionsReducer = (state = { missions: [] }, action) => {
  switch (action.type) {
    case MISSION_START:
      return {
        ...state,
        loading: true,
      };
    case MISSION_FAILED:
      return {
        ...state,
        loading: false,
        error: action.payload,
      };`
      
- thunk folder will keep the API implementations, fetch some information from the API link.
 
Example

`export const loadMissionsData = () => (dispatch) => {
  dispatch(loadingStart());
  fetchData()
    .then((res) => res.json())
    .then((result) => dispatch(loadMission(result)))
    .catch((err) => dispatch(manageFailure(err.message)));
};`

Please, let us try to split it out into different files to keep our code clean and easy to maintain.